content/s14_user_profiles/s14_p03_user_avatars.md

```diff
@@ -0,0 +1,147 @@
+Don't you want your users to have avatars? And no, I don't mean airbenders or aliens, but...
+
+Unicorns.
+
+![](https://images.unsplash.com/photo-1516486915195-358a5c2b2395?ixlib=rb-1.2.1&auto=format&fit=crop&w=634&q=80)
+
+No, I'm not going crazy. Anyway, in this lesson, you'll learn about how you can give each user a unique avatar, or a profile pic they can use to represent themselves.
+
+### Unicornify
+
+A social media site can be improved with some pictures, especially for users. Because each user a is a unique snowflake. <a href="https://unicornify.pictures/" target="_blank">Unicornify</a> is a tiny service written by Benjamin Dumke-von der Ehe. Given an MD5 hash, the service will give you a unique image of a unicorn, like this one:
+
+![](https://unicornify.pictures/avatar/7daf6c79d4802916d8516266e74850af?s=128)
+
+Unicornify is inspired by <a href="https://en.gravatar.com/" target="_blank">Gravatar</a>, which is the most popular avatar service. It associates each user's email with a unique avatar, and user's can change their avatar by making an account on the Gravatar website.
+
+If you haven't visited the Unicornify website yet, it gives these instructions:
+
+```
+URL format: https://unicornify.pictures/avatar/$HASH?s=$SIZE
+
+$HASH is any hexadecimal number, up to 64 digits.
+$SIZE is the requested image size in pixels.
+It should be a power of 2 between 32 and 128 (inclusive).
+```
+
+### hashlib
+
+To make these MD5 hashes, we'll use each user's email as the input. Let's try an example email with the Python standard library `hashlib`:
+
+```bash
+(env) $ python
+>>> import hashlib
+>>> hashlib.md5('paul@example.com'.encode('utf-8')).hashdigest()
+Traceback (most recent call last):
+  File "<stdin>", line 1, in <module>
+AttributeError: '_hashlib.HASH' object has no attribute 'hashdigest'
```

Comment:
If you keep this `AttributeError` then I'd say explain why it's in there. otherwise just take it out and do the `.hexdigest()` right away (makes more sense here IMO)